3.0.1 build issue on Windows

Compiling php_http.dll on Windows (x64, PHP7.0.11, nts, VC14) leads to this error with the 3.0.1RC1 release or the sources from git head
```
c:\php-sdk\php70dev\ext\http\src\php_http_api.h(116): error C2016: C requires that a struct
 or union has at least one member (compiling source file ext\http\src\php_http_options.c)
```

--- a/config.w32
+++ b/config.w32
@@ -90,6 +90,17 @@ if (PHP_HTTP != "no") {
 		ADD_EXTENSION_DEP("http", "iconv", true);
 	}
 
+	if (PHP_INTL != "no") {
+		ADD_EXTENSION_DEP("http", "intl", true);
+		AC_DEFINE("HAVE_IDNA2008", 1, "");
+		AC_DEFINE("HAVE_UIDNA_IDNTOASCII", 1, "");
+		AC_DEFINE("HAVE_UIDNA_NAMETOASCII_UTF8", 1, "");
+		AC_DEFINE("PHP_HTTP_HAVE_LIBICU", 1, "");
+		AC_DEFINE("PHP_HTTP_HAVE_IDNA2003", 1, "");
+		AC_DEFINE("PHP_HTTP_HAVE_IDNA2008", 1, "");
+		ADD_FLAG("LIBS_HTTP", "icuuc.lib icudt.lib icuin.lib icuio.lib icule.lib iculx.lib");
+	}
+
 	if (PHP_CURL != "no") {
 		ADD_EXTENSION_DEP("http", "curl", true);
 	}

--- a/config.w32
+++ b/config.w32
@@ -103,7 +103,7 @@ if (PHP_HTTP != "no") {
 			ADD_EXTENSION_DEP("http", "propro", true) &&
 			ADD_EXTENSION_DEP("http", "raphf", true) &&
 			CHECK_LIB("winmm.lib", "http", PHP_HTTP)) {
-		AC_DEFINE("PHP_HTTP_HAVE_CURL", 1, "Have CURL library");
+		AC_DEFINE("PHP_HTTP_HAVE_LIBCURL", 1, "Have CURL library");
 		AC_DEFINE("PHP_HTTP_HAVE_SSL", 1, "Have SSL");
 		AC_DEFINE("PHP_HAVE_CURL_MULTI_STRERROR", 1, "");
 		AC_DEFINE("PHP_HAVE_CURL_SHARE_STRERROR", 1, "");

--- a/config.w32
+++ b/config.w32
@@ -65,6 +65,7 @@ if (PHP_HTTP != "no") {
 
 	if (CHECK_HEADER_ADD_INCLUDE('zlib.h', 'CFLAGS_HTTP', '..\\zlib;' + php_usual_include_suspects)) {
 		AC_DEFINE('HTTP_HAVE_ZLIB', 1, "Have zlib library");
+		AC_DEFINE('PHP_HTTP_HAVE_ZLIB', 1, "");
 		ADD_FLAG("LDFLAGS_HTTP", "/FORCE:MULTIPLE");
 	} else {
 		WARNING("zlib encoding functions not enabled; libraries and headers not found");
@@ -74,6 +75,7 @@ if (PHP_HTTP != "no") {
 		var f;
 
 		if ((f = check_for_pecl_ext("hash")) || (f = check_for_main_ext("hash"))) {
+			AC_DEFINE("PHP_HTTP_HAVE_EXT_HASH", 1, "");
 			ADD_FLAG("CFLAGS_HTTP", '/I "' + f + '" /DHTTP_HAVE_PHP_HASH_H=1');
 			ADD_EXTENSION_DEP("http", "hash", true);
 		}
@@ -84,6 +86,7 @@ if (PHP_HTTP != "no") {
 	}
 
 	if (PHP_ICONV != "no") {
+		AC_DEFINE("PHP_HTTP_HAVE_EXT_ICONV", 1, "");
 		ADD_EXTENSION_DEP("http", "iconv", true);
 	}
 
@@ -105,6 +108,11 @@ if (PHP_HTTP != "no") {
 			CHECK_LIB("winmm.lib", "http", PHP_HTTP)) {
 		AC_DEFINE("PHP_HTTP_HAVE_LIBCURL", 1, "Have CURL library");
 		AC_DEFINE("PHP_HTTP_HAVE_SSL", 1, "Have SSL");
+		AC_DEFINE("PHP_HTTP_HAVE_LIBCURL_SSL", 1);
+		AC_DEFINE("PHP_HTTP_HAVE_LIBCURL_OPENSSL", 1);
+		AC_DEFINE("PHP_HTTP_HAVE_EXT_PROPRO", 1, "");
+		AC_DEFINE("PHP_HTTP_HAVE_EXT_RAPHF", 1, "");
+		AC_DEFINE("PHP_HTTP_HAVE_CLIENT", 1, "Have HTTP client support");
 		AC_DEFINE("PHP_HAVE_CURL_MULTI_STRERROR", 1, "");
 		AC_DEFINE("PHP_HAVE_CURL_SHARE_STRERROR", 1, "");
 		AC_DEFINE("PHP_HAVE_CURL_EASY_STRERROR", 1, "");
@@ -113,7 +121,6 @@ if (PHP_HTTP != "no") {
 		AC_DEFINE("PHP_HAVE_CURL_FORMGET", 1, "");
 		AC_DEFINE("PHP_HAVE_CURL_MULTI_SETOPT", 1, "");
 		AC_DEFINE("PHP_HAVE_CURL_MULTI_TIMEOUT", 1, "");
-		AC_DEFINE("PHP_HTTP_HAVE_CLIENT", 1, "Have HTTP client support");
  		ADD_FLAG("CFLAGS_HTTP", "/I" + configure_module_dirname + " ");
 
 		if (CHECK_HEADER_ADD_INCLUDE("event2/event.h", "CFLAGS_HTTP") &&
@@ -121,9 +128,9 @@ if (PHP_HTTP != "no") {
 			CHECK_LIB("libevent_core.lib", "http", PHP_HTTP) &&
 			CHECK_LIB("libevent_extras.lib", "http", PHP_HTTP)) {
 
-			AC_DEFINE("PHP_HTTP_HAVE_EVENT", 1);
-			AC_DEFINE("PHP_HTTP_HAVE_EVENT2", 1);
-			AC_DEFINE("PHP_HTTP_EVENT_VERSION", "2.0.21 or greater");
+			AC_DEFINE("PHP_HTTP_HAVE_LIBEVENT", 1);
+			AC_DEFINE("PHP_HTTP_HAVE_LIBEVENT2", 1);
+			AC_DEFINE("PHP_HTTP_LIBEVENT_VERSION", "2.0.21 or greater");
 		}
 	} else {
 		WARNING("curl convenience functions not enabled; libraries and headers not found");

